src/main.rs

```diff
@@ -67,11 +67,13 @@ async fn get_object(
         if let Ok(r) = range_header.to_str() {
             if let Some(bytes_range) = r.strip_prefix("bytes=") {
                 if let Some((start, end)) = bytes_range.split_once('-') {
-                    if let (Ok(s), Ok(e)) = (start.parse::<u64>(), end.parse::<u64>()) {
+                    if let Ok(s) = start.parse::<u64>() {
                         range_start = s;
-                        range_end = e;
-                        range = Some(r.to_string());
-                        is_range_request = true;
+                        if end.is_empty() || end.parse::<u64>().is_ok() {
+                            range = Some(r.to_string());
+                            is_range_request = true;
+                            range_end = end.parse::<u64>().unwrap_or(u64::MAX);
```

Comment:
I think then calculating the range_end using range_start and content_length should do it at the time of adding the response header.
range_start + content_length.parse::<u64>().unwrap() - 1.

Reply:
Sure, that will work if the end range wasn’t defined but it won’t return the correct header response if the end range *was* defined

Reply:
why not always do that and ditch the variable range_end!

Reply:
Sorry, I missed that you added `range_start` to the value, that would be incorrect. In the case of an undefined `range_end` then the `range_end` in the response should be the content length minus 1. In the case that the `range_end` is defined then the `range_end` in the response should be equal to what was defined.

Reply:
Sorry, I missed that you added range_start to the value, that would be incorrect. Whats incorrect?

Reply:
`range_end = range_start + content_length.parse::<u64>().unwrap() - 1` would be incorrect

Assuming a requested file with a `Content-Length` of `999` bytes here's how two different range requests should be handled:

* `Range` header in the request set to `bytes=99-`
  *  `Content-Range` header in the response should be set to `bytes 99-998/999`
* `Range` header in the request set to `bytes=0-200`
  *  `Content-Range` header in the response should be set to `bytes 0-200/999`

Reply:
@navdeepg2021 I'm not sure if you understood correctly what I was trying to point out.

The way your updated code works is such that the `Content-Range` header returns a value which is effectively `bytes {range_start}-{range_start+content_length-1}/{content_length}`. This is **incorrect**.

See my previous comment for how the responses should be constructed for two different cases, one where the user initiates the range request with an unspecified HTTP end range and one where the user initiates a request with a specified HTTP end range.

It absolutely would help you out here to write a test first which tests that that values for that header match the values expected given a set of parameters. You can use the two cases and their values in my previous comment as input parameters and expected outputs.

As a note, in case you are having difficulties writing test cases in Rust, I have found that writing tests is a great use case for AI tools. It helps out a ton on generating the boiler plate code required and lets you focus on just the important bits. Claude 3.5 Sonnet has been really great for me when working with Rust.

Reply:
see the difference between content_length variable. one is res.content_length and another is full content_length!

Reply:
Ah, my bad! This looks good then! Ideally I really would like to see tests written for fixes to bugs like this just to really make sure we don't have any regressions but I will approve this case